includes/abstracts/abstract-wc-rest-terms-controller.php

```diff
@@ -533,6 +533,14 @@ public function delete_item( $request ) {
 		}
 
 		$term = get_term( (int) $request['id'], $taxonomy );
+		// Get default category id.
+		$default_category_id = absint( get_option( 'default_product_cat', 0 ) );
+
+		// Prevent deleting the default product category.
+		if ( $default_category_id === $term ) {
+			return new WP_Error( 'woocommerce_rest_default_product_cat_cannot_delete', __( 'Default product category cannot be deleted.', 'woocommerce' ), array( 'status' => 500 ) );
```

Comment:
Also it would be best to use the existing `woocommerce_rest_cannot_delete` error group here instead of `woocommerce_rest_default_product_cat_cannot_delete`.